odoo/addons/base/tests/test_views.py

```diff
@@ -674,6 +674,45 @@ def test_branding_inherit(self):
             second.get('data-oe-id'),
             "second should come from the extension view")
 
+    def test_branding_inherit_replace_node(self):
+        view1 = self.View.create({
+            'name': "Base view",
+            'type': 'qweb',
+            'arch': """<hello>
+                <world></world>
+                <world><t t-esc="hello"/></world>
+                <world></world>
+            </hello>
+            """
+        })
+        view2 = self.View.create({
+            'name': "Extension",
+            'type': 'qweb',
+            'inherit_id': view1.id,
+            'arch': """<xpath expr="/hello/world[1]" position="replace">
+                <world>Is a ghetto</world>
+            </xpath>
+            """
+        })
+
+        arch_string = view1.with_context(inherit_branding=True).read_combined(['arch'])['arch']
+
+        arch = etree.fromstring(arch_string)
+        self.View.distribute_branding(arch)
+
+        for i in [1, 3]:
+            xpath = '/hello[1]/world[%s]' % i
+            [initial] = arch.xpath(xpath)
+            self.assertEqual(
+                xpath,
+                initial.get('data-oe-xpath'),
+                "initial should come from the root view")
```

Comment:
I think that this `assert` is testing what it should for the third world (the only really useful assert in the test is that one) but it is not correct to test the first world with it. As the first one comes from the inherit view, I think its oe-xpath is very different (something like `/xpath[1]/world[1]`) ?